### ISSUE TITLE
Missing function in documentation

### ISSUE THREAD

**Issue by groue** [realm/jazzy#266]
Hello,

In http://cocoadocs.org/docsets/GRDB.swift/0.7.0/Extensions/Database.html, the documentation for this function is missing: https://github.com/groue/GRDB.swift/blob/v0.7.0/GRDB/RowModel.swift#L684-L698.

I don't quite know why.

Thank you!

**Comment by jpsim** [comment]
It looks like it's there? http://cocoadocs.org/docsets/GRDB.swift/0.7.0/Extensions/Database.html#/s:FC4GRDB8Database5fetchERR

**Comment by groue** [comment]
No, it is not the same signature, even if they look pretty similar.

**Comment by jpsim** [comment]
Ah, yes you're right... this need some more serious debugging.

**Comment by jpsim** [comment]
Can you please share the parameters you're passing to jazzy? Is it possible you're passing `-dry-run` to xcodebuild? That could lead to this sort of error.

**Comment by groue** [comment]
No, not really. It's just cocoapods documentatilon generation, and I don't quite know how they generate documentation with jazzy (maybe you could get in touch with them).

Yet, your initial confusion may put you on the right track; both functions look like they are the same, but they are not. I'm not sure any parameter has anything to do with it.

**Comment by groue** [comment]
For reference, here are the two similar functions (only one is documented by jazzy):

[Function 1](https://github.com/groue/GRDB.swift/blob/v0.7.0/GRDB/Database.swift#L298-L311) (documented), [function 2](https://github.com/groue/GRDB.swift/blob/v0.7.0/GRDB/RowModel.swift#L684-L698) (not documented):

```swift
public func fetch<Value: DatabaseValueConvertible>(type: Value.Type, _ sql: String, arguments: QueryArguments? = nil) -> AnySequence<Value?>
public func fetch<RowModel: GRDB.RowModel>(type: RowModel.Type, _ sql: String, arguments: QueryArguments? = nil) -> AnySequence<RowModel>
```

We have the same issue here: http://cocoadocs.org/docsets/GRDB.swift/0.7.0/Classes/DatabaseQueue.html

[Function 1](https://github.com/groue/GRDB.swift/blob/v0.7.0/GRDB/DatabaseQueue.swift#L71-L91) (documented), [function 2](https://github.com/groue/GRDB.swift/blob/v0.7.0/GRDB/DatabaseQueue.swift#L93-L113) (not documented):

```swift
public func inDatabase(block: (db: Database) throws -> Void) rethrows
public func inDatabase<R>(block: (db: Database) throws -> R) rethrows -> R
```

I really don't think parameters have anything to do here. It really looks like there is a bug in function identification that triggers when signatures are very close.

**Comment by jpsim** [comment]
It could very well be failing because Jazzy only works on Swift 1.0, 1.1 and 1.2 at the moment. Maybe try generating your docs with #261?

**Comment by groue** [comment]
It looks like Cocoadocs.org uses jazzy 0.2.3 (https://github.com/CocoaPods/cocoadocs.org/blob/64cfbd344b6bdf0e8f15847ef474247a4e4b9559/Gemfile.lock#L76).

Yet the two following functions don't use Swift 2 features, and yet only one is documented:

```swift
public func fetch<Value: DatabaseValueConvertible>(type: Value.Type, _ sql: String, arguments: QueryArguments? = nil) -> AnySequence<Value?>
public func fetch<RowModel: GRDB.RowModel>(type: RowModel.Type, _ sql: String, arguments: QueryArguments? = nil) -> AnySequence<RowModel>
```

I strongly believe that there is a bug in the way jazzy identifies functions. Do you think it could be a bug in https://github.com/jpsim/SourceKitten?

**Comment by jpsim** [comment]
I think it's either a bug in SourceKitten, or the fact that this is a Swift 2 code base. Jazzy itself actually has no knowledge of Swift parsing, all of which is done by SourceKitten.

**Comment by groue** [comment]
OK. Do you know who is responsible of function identification ? I mean, does SourceKitten exposes an AST that jazzy visits in order to extract functions, or does SourceKitten expose a pre-built list of functions declared in a swift file?

**Comment by jpsim** [comment]
> SourceKitten expose a pre-built list of functions declared in a swift file?

This. See [Source/SourceKittenFrameworkTests/Fixtures/Bicycle.json](https://github.com/jpsim/SourceKitten/blob/master/Source/SourceKittenFrameworkTests/Fixtures/Bicycle.json) for an example SourceKitten doc output in the format that jazzy consumes. It's generated from this [Source/SourceKittenFrameworkTests/Fixtures/Bicycle.swift](https://github.com/jpsim/SourceKitten/blob/master/Source/SourceKittenFrameworkTests/Fixtures/Bicycle.swift) file.

**Comment by mkieselmann** [comment]
Hey guys,
i have the same issue with overloaded functions. Only the first declaration of functions with the same name but different parameters appears in the documentation.

Jazzy outputs the following errors while generating the docs:
```
Found conflicting type declarations with the same name, which may indicate a build issue or a bug in Jazzy: instance method persist(_:), instance method persist(_:)
Found conflicting type declarations with the same name, which may indicate a build issue or a bug in Jazzy: instance method persist(_:), instance method persist(_:)
```

Setting:
jazzy 0.6.2
Swift 2.2
The following command is used in bash script to build a swift framework with the target name "$componentName".
```
jazzy -o "$outputDir" \  
            --clean \
            --source-directory "$pathToSdkRoot/$1/$2" \
            --readme "$readmeFilePath" \
            --module "$componentName" \
```

**Comment by jpsim** [comment]
@mkieselmann could you share a minimal project that reproduces this issue so I can take a closer look? Thanks!

**Comment by mkieselmann** [comment]
@jpsim thanks for pointing me to creating a minimal project because I found the issue during the creation.
Our setup consists of one Xcode workspace and a couple of interdependent iOS framework within that workspace. A bash script generates a jazzy documentation for each framework project. Since I used the `--module` argument, jazzy calls xcodebuild without the workspace and the dependencies could not be resolved.
I had to use the following jazzy argument:
`jazzy -o "$outputDir" \
            --xcodebuild-arguments -workspace,"$workspaceFilePath",-scheme,"$componentName" \
            --readme "$readmeFilePath" \
            --config "$subcategoriesPath" \
            --module "$componentName" `
